elasticsearch/client.go

```diff
@@ -135,3 +142,65 @@ func newV8Client(apikey, user, pwd string, addresses []string, transport http.Ro
 		Transport: transport,
 	})
 }
+
+type JSONResponse struct {
+	content io.ReadCloser
+	err     error
+}
+
+func (r JSONResponse) DecodeTo(i interface{}) error {
+	if r.err != nil {
+		return r.err
+	}
+	bs, err := ioutil.ReadAll(r.content)
+	if err != nil {
+		return err
+	}
+	err = json.Unmarshal(bs, i)
```

Comment:
can lighten the footprint here with `json.NewDecoder(r.content).Decode(&i)`.  Maybe this whole function can be replaced if the error is check before calling?